### DISCUSSION TITLE
Casting bytes into column in dataset

### DISCUSSION THREAD

**Discussion by AvoyDatta** [huggingface/datasets#5593] [Q&A]
Hi ,

I am solving an ASR task, and have my audio files as`byte` strings. How can I cast them using the Audio dataclass (`datasets.Audio`)? Thanks!

**Comment by mariosasko** [comment]
Hi! I've opened a [PR](https://github.com/huggingface/datasets/pull/5644) that adds a "kernel" to allow this kind of cast.

In the meantime, you can perform the cast as follows:
```python
def convert_audio_bytes_to_dict(batch):
    batch["audio"] = [{"bytes": audio_bytes} for audio_bytes in batch["audio"]]
    return batch

# convert the column type from Value("binary") to Audio()
features = ds.features
features = features.copy()
features["audio"] = datasets.Audio()

ds = ds.map(convert_audio_bytes_to_dict, batched=True, features=features)
```
